### ISSUE TITLE
Can we train the model on a 4 X 40G gpus or even fewer?

### ISSUE THREAD

**Issue by 652994331** [agentica-project/deepscaler#67]
Hi , really appreciate our work, it's amazing. I have a question ,  our team does not have enough gpu , just some 40G GPUS. is it possible for us to train the RL process , or maybe just 8k steps , on 4 x 40G gpus， or even fewer , maybe 1 or 2 gpus? . we really hope we can at least do something with this kind of low resources... :( . Thank you

**Comment by GangLii** [comment]
Setting actor_rollout_ref.actor.use_dynamic_bsz=True and actor_rollout_ref.actor.ppo_max_token_len_per_gpu=28672 might work 40G gpus. But be careful about the learning rate as the learning rate is not properly scaled in the repo when use_dynamic_bsz=True.

**Comment by ftgreat** [comment]
original 8k scripts can run on a 8 X 40G gpus

**Comment by 652994331** [comment]
Thank you, is it possible to run on 4*40G gpus
If we modify some training parameters?



发自我的iPhone


------------------ Original ------------------
From: ldwang ***@***.***&gt;
Date: Wed,Mar 26,2025 0:04 PM
To: agentica-project/deepscaler ***@***.***&gt;
Cc: 652994331 ***@***.***&gt;, Author ***@***.***&gt;
Subject: Re: [agentica-project/deepscaler] Can we train the model on a 4 X 40Ggpus or even fewer? (Issue #67)

**Comment by 652994331** [comment]
@ftgreat Thank you, is it possible to run on 4*40G gpus
If we modify some training parameters?

**Comment by cht619** [comment]
> Hi [@ftgreat](https://github.com/ftgreat), thank you! I’m wondering if it's possible to run DeepScaleR training on 4 × 40GB GPUs by adjusting some training parameters.

If it’s convenient for you, could you please point out which parameters we might need to modify (e.g., batch size, gradient accumulation, sequence length, etc.)?

**Comment by 652994331** [comment]
@cht619  I figured it out , modifying response_length , mini_batch_szie and micro batch_size can provide help here. although it's very slow, it worked. However, there is a concern about the performance..

**Comment by ftgreat** [comment]
> > Hi [@ftgreat](https://github.com/ftgreat), thank you! I’m wondering if it's possible to run DeepScaleR training on 4 × 40GB GPUs by adjusting some training parameters.
> 
> If it’s convenient for you, could you please point out which parameters we might need to modify (e.g., batch size, gradient accumulation, sequence length, etc.)?

I just use the original script of 8k training without any changes.

**Comment by ftgreat** [comment]
> [@ftgreat](https://github.com/ftgreat) Thank you, is it possible to run on 4*40G gpus If we modify some training parameters?

Maybe all batch_size configs.

**Comment by tanghui315** [comment]
Is it possible to run 32B models with  4*40G gpus  ?

**Comment by cht619** [comment]
4*48GPU, OOM
